ios/Classes/TiNfcNfcAdapterProxy.m

```diff
@@ -0,0 +1,77 @@
+/**
+ * Ti.NFC
+ * Copyright (c) 2009-2017 by Axway Appcelerator. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#if IS_IOS_11
+
+#import "TiNfcNfcAdapterProxy.h"
+#import "TiNfcNdefMessageProxy.h"
+#import "TiNfcUtilities.h"
+#import "TiUtils.h"
+
+@implementation TiNfcNfcAdapterProxy
+
+#pragma mark Internal
+
+- (NFCNDEFReaderSession *)nfcSession
+{
+  if (_nfcSession == nil) {
+    _nfcSession = [[NFCNDEFReaderSession alloc] initWithDelegate:self
+                                                           queue:nil
+                                        invalidateAfterFirstRead:[TiUtils boolValue:[self valueForKey:@"invalidateAfterFirstRead"] def:NO]];
+  }
+
+  return _nfcSession;
+}
+
+#pragma mark Public API's
+
+- (void)begin:(id)unused
+{
+  [[self nfcSession] beginSession];
+}
+
+- (void)invalidate:(id)unused
+{
+  [[self nfcSession] invalidateSession];
+  _nfcSession = nil;
+}
+
+- (void)setOnNdefDiscovered:(KrollCallback *)callback
+{
+  [self replaceValue:callback forKey:@"onNdefDiscovered" notification:NO];
+  _ndefDiscoveredCallback = callback;
+}
+
+#pragma mark NFCNDEFReaderSessionDelegate
+
+- (void)readerSession:(NFCNDEFReaderSession *)session didDetectNDEFs:(NSArray<NFCNDEFMessage *> *)messages
+{
+  NSMutableArray *result = [NSMutableArray arrayWithCapacity:[messages count]];
+
+  for (NFCNDEFMessage *message in messages) {
+    [result addObject:[[TiNfcNdefMessageProxy alloc] _initWithPageContext:[self pageContext]
```

Comment:
What is use of result? It is not getting used anywhere in this function.

Reply:
It is returned in the same method and `- (NSArray<TiNfcNdefRecordProxy *> *)records` returns to the proxy as `records`.

Reply:
I think you are talking about TiNfcNdefMessageProxy class. But above method is delegate method of TiNfcNfcAdapterProxy class. You are passing -
[_ndefDiscoveredCallback call:@[ @{@"messages" : [TiNfcUtilities arrayFromNDEFMessages:messages]} ]            thisObject:self];
Corresponding to "messages"  key I think you have to pass result. If I understand right, I think there is no use of TiNfcUtilities class. Can you please once review  delegate method -
+- (void)readerSession:(NFCNDEFReaderSession *)session didDetectNDEFs:(NSArray<NFCNDEFMessage *> *)messages

of  TiNfcNfcAdapterProxy class.  Thanks!

Reply:
Checking it, the loop is indeed not used, it was moved to the utilities. But the `TiNfcUtilities` are still used to bridge the messages to proxies. I updated the PR to remove the unused code!

Reply:
If you process the messages received in  callback using TiNfcUtilities then TiNfcNdefMessageProxy  and TiNfcNdefRecordProxy class is of no use.

Reply:
You are right! Fixed that.